File: blog-post-2/programjava/ArmStrong.java
package programjava;

import java.util.Scanner;

public class ArmStrong {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter number: ");
        int number = scanner.nextInt();
        scanner.close();
        int temp = number;
        int count = 0;
        int result = 0;
        while(number > 0 ){
            count++;
            number = number/10;
        }

        number = temp;

        while(number > 0){
            final int reminder = number % 10;
            result += Math.pow(reminder, count);
            number /= 10;
        }
        if(temp == result){
            System.out.println("Armgstrong number");
        }
        else{
            System.out.println("Not armgstrong number");
        }
    }
}

File: blog-post-2/c/ArmStrong.c
#include &lt;stdio.h&gt;
int main()
{
    int number, temp, count, result, reminder, pow = 1;
    printf("Enter number: ");
    scanf("%d", &number);
    temp = number;
    count = 0;
    result = 0;
    while (number &gt; 0)
    {
        count++;
        number = number / 10;
    }

    number = temp;

    while (number &gt; 0)
    {
        reminder = number % 10;
        while (count &gt; 0)
        {
            pow *= reminder;
            count--;
        }
        result += reminder;
        number /= 10;
        pow = 1;
    }
    if (temp == result)
    {
        printf("Armgstrong number");
    }
    else
    {
        printf("Not armgstrong number");
    }
    return 0;
}